presto-elasticsearch/src/main/java/com/facebook/presto/elasticsearch/ElasticsearchConnectorModule.java

```
@@ -0,0 +1,152 @@
+    {
+        binder.bind(ElasticsearchConnector.class).in(Scopes.SINGLETON);
+        binder.bind(ElasticsearchMetadata.class).in(Scopes.SINGLETON);
+        binder.bind(ElasticsearchSplitManager.class).in(Scopes.SINGLETON);
+        binder.bind(ElasticsearchRecordSetProvider.class).in(Scopes.SINGLETON);
+
+        configBinder(binder).bindConfig(ElasticsearchConnectorConfig.class);
+
+        jsonBinder(binder).addDeserializerBinding(Type.class).to(TypeDeserializer.class);
+        jsonCodecBinder(binder).bindJsonCodec(ElasticsearchTableDescription.class);
+
+        binder.install(new DecoderModule());
+    }
+
+    @Singleton
+    @Provides
+    public static ElasticsearchClient createElasticsearchClient(ElasticsearchConnectorConfig config, JsonCodec<ElasticsearchTableDescription> tableDescriptionCodec)
+    {
+        ImmutableMap.Builder<SchemaTableName, ElasticsearchTableDescription> builder = ImmutableMap.builder();
+        try {
+            for (File file : listFiles(config.getTableDescriptionDir())) {
+                if (file.isFile() && file.getName().endsWith(".json")) {
+                    ElasticsearchTableDescription table = tableDescriptionCodec.fromJson(readAllBytes(file.toPath()));
+                    String schemaName = firstNonNull(table.getSchemaName(), config.getDefaultSchema());
+                    builder.put(new SchemaTableName(schemaName, table.getTableName()), table);
+                }
+            }
+
+            Map<SchemaTableName, ElasticsearchTableDescription> tableDefinitions = builder.build();
+
+            builder = ImmutableMap.builder();
+            for (String definedTable : config.getTableNames()) {
+                SchemaTableName tableName;
+                try {
+                    tableName = parseTableName(definedTable);
+                }
+                catch (IllegalArgumentException iae) {
+                    tableName = new SchemaTableName(config.getDefaultSchema(), definedTable);
```

Why do we silently use the default schema if the tables specified in the config are not valid? 
I think it may be better to fail loudly here as it's possible that the user made a mistake configuring the system and then the default schema will be used, which may not be what the user wants.

---

get it, will fix